pkg/pillar/cmd/volumemgr/create.go

```diff
@@ -33,13 +37,27 @@ func createVdiskVolume(ctx *volumemgrContext, status types.VolumeStatus,
 
 	created := false
 
+	persistFsType := vault.ReadPersistType()
+
 	// this is the target location, where we expect the volume to be
 	filelocation := status.PathName()
-	if _, err := os.Stat(filelocation); err == nil {
-		errStr := fmt.Sprintf("Can not create %s for %s: exists",
-			filelocation, status.Key())
-		log.Error(errStr)
-		return created, "", errors.New(errStr)
+
+	if persistFsType == types.PersistZFS {
+		zvolName := status.ZVolName()
+		args := []string{"/hostfs", "zfs", "get", "all", zvolName}
+		if _, err := base.Exec(log, vault.ZfsPath, args...).CombinedOutput(); err == nil {
```

Comment:
Is there some golang package (with a reasonable license) which can be used to avoid execing binaries? Under memory pressure the fork/exec can be quite slow.

Reply:
There is a few @eriknordmark -- but almost all of them are just thin wrappers on top of CLI execution unfortunately. Klara folks have proposed a native bindings into Go, but that's a bit away timing-wise.